src/System.Security.Cryptography.OpenSsl/tests/EcDsaOpenSslImportExportTests.cs

```diff
@@ -0,0 +1,339 @@
+
+                    Assert.True(
+                        curveDef.IncludePrivate && param2.D.Length > 0 ||
+                        !curveDef.IncludePrivate && param2.D == null);
+                    Assert.Equal(curveDef.CurveType, curve.CurveType);
+                    Assert.Equal(curveDef.KeyLength, newEc.KeySize);
+
+                    // values should be the same
+                    IsEqual(param, param2);
+                    Assert.True(IsEqual(param.D, param2.D));
+                }
+            }
+        }
+
+        private static void OutputExplicitCurveForDebugging(string prefix, ECParameters paramaters)
+        {
+            ECExplicitCurve curve = (ECExplicitCurve)paramaters.Curve;
+
+            System.Diagnostics.Debug.WriteLine("CurveType:" + curve.CurveType.ToString());
+            System.Diagnostics.Debug.WriteLine("qx{0}:{1}", prefix, paramaters.Q.X.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("qy{0}:{1}", prefix, paramaters.Q.Y.ByteArrayToHex());
+            if (paramaters.D != null)
+                System.Diagnostics.Debug.WriteLine("d{0}:{1}", prefix, paramaters.D.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("a{0}:{1}", prefix, curve.A.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("b{0}:{1}", prefix, curve.B.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("gx{0}:{1}", prefix, curve.G.X.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("gy{0}:{1}", prefix, curve.G.Y.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("h{0}:{1}", prefix, curve.H.ByteArrayToHex());
+            System.Diagnostics.Debug.WriteLine("n{0}:{1}", prefix, curve.N.ByteArrayToHex());
+            if (curve.S != null)
+                System.Diagnostics.Debug.WriteLine("s{0}:{1}", prefix, curve.S.ByteArrayToHex());
+
+            if (curve is ECPrimeCurve)
+            {
+                System.Diagnostics.Debug.WriteLine("p{0}:{1}", prefix, ((ECPrimeCurve)curve).P.ByteArrayToHex());
+            }
+            else
+            {
+                System.Diagnostics.Debug.WriteLine("m{0}:{1}", prefix, ((ECCharacteristic2Curve)curve).M.ByteArrayToHex());
+            }
+        }
+
+        private static void IsEqual(ECParameters p1, ECParameters p2)
+        {
+            ComparePublicKey(p1.Q, p2.Q);
+            CompareCurve(p1.Curve, p2.Curve);
+        }
+
+        private static void ComparePublicKey(ECPoint q1, ECPoint q2, bool isEqual = true)
+        {
+            Assert.Equal(isEqual, IsEqual(q1.X, q2.X));
+            Assert.Equal(isEqual, IsEqual(q1.Y, q2.Y));
+        }
+
+        private static void CompareCurve(ECCurve c1, ECCurve c2)
+        {
+            Assert.True((c1 == null && c2 == null) || (c1 != null && c2 != null));
+            if (c1 is ECNamedCurve)
+            {
+                var curve1 = (ECNamedCurve)c1;
+                var curve2 = (ECNamedCurve)c1;
```

Comment:
Copy-paste error from cng. Note I still plan on combining common tests and helper methods like this between cng and openssl